### PR TITLE
D9

### PR DESCRIPTION
**Pull request by alexkras**
Many modules are not ready yet, like default_content. 
current build allow to see working d9 site with sdd profile which mostly empty.

### CODE REVIEW

File: composer.json [hunk 1]
```diff
@@ -1,159 +1,116 @@
...
-    {
-      "type": "package",
-      "package": {
-        "name": "ckeditor/panelbutton",
-        "version": "4.10.1",
-        "type": "drupal-library",
-        "dist": {
-          "url": "https://download.ckeditor.com/panelbutton/releases/panelbutton_4.10.1.zip",
-          "type": "zip"
+    "name": "skilld-labs/sdc",
```
Comment (piggito) [inline composer.json:2]: Please keep same tabbing as on previous composer.json file to easily identify the changes

File: Makefile [hunk 2]
```diff
@@ -90,10 +90,10 @@ else
 	$(call php, drush si $(PROFILE_NAME) --db-url=$(DB_URL) --account-name=$(ADMIN_NAME) --account-mail=$(ADMIN_MAIL) --account-pass=$(ADMIN_PW) -y --site-name="$(SITE_NAME)" --site-mail="$(SITE_MAIL)" install_configure_form.site_default_country=FR install_configure_form.date_default_timezone=Europe/Paris)
 endif
 ifneq ($(strip $(MODULES)),)
```
Comment (piggito) [inline Makefile:92]: In other PR https://github.com/skilld-labs/skilld-docker-container/pull/232#discussion_r406734049 @andypost mentioned we are getting rid of default_content so IMO we should remove this part

Mb we need a new issue to explore new ways to deal with content import.

File: scripts/composer/ScriptHandler.php [hunk 3]
```diff
@@ -12,6 +12,7 @@
 use DrupalFinder\DrupalFinder;
 use Symfony\Component\Filesystem\Filesystem;
 use Webmozart\PathUtil\Path;
+use Drupal\Core\Site\Settings;
```
Comment (piggito) [inline scripts/composer/ScriptHandler.php:15]: Please keep imports ordered

File: web/modules/custom/project_default_content/project_default_content.info.yml [hunk 4]
```diff
@@ -4,6 +4,7 @@ package: Custom
 
 type: module
 core: 8.x
+core_version_requirement: ^8||^9
 
 dependencies:
   - default_content:default_content
```
Comment (piggito) [inline web/modules/custom/project_default_content/project_default_content.info.yml:10]: As mentioned before, @andypost idea is to get rid of default_content so IMO we should also remove this module until we decide on the new way to import content. We can add it back later if needed.

File: web/profiles/sdd/sdd.info.yml [hunk 5]
```diff
@@ -1,21 +1,21 @@
...
-core: 8.x
+core_version_requirement: ^9
 
 dependencies:
   - automated_cron
   - big_pipe
   - block
   - block_content
-  - block_content_permissions
+#  - block_content_permissions
```
Comment (piggito) [inline web/profiles/sdd/sdd.info.yml:11]: Please don't comment here, remove the lines and we can create issues to migrate those modules to d9

File: web/profiles/sdd/config/install/block.block.seven_breadcrumbs.yml [hunk 6]
```diff
@@ -1,19 +0,0 @@
-langcode: en
```
Comment (piggito) [inline web/profiles/sdd/config/install/block.block.seven_breadcrumbs.yml:1]: You removed all of config/install folder, need to add them back with changes included in d9

File: web/profiles/sdd_core/config/install/block.block.seven_breadcrumbs.yml [hunk 7]
```diff
@@ -1,19 +0,0 @@
-langcode: en
```
Comment (piggito) [inline web/profiles/sdd_core/config/install/block.block.seven_breadcrumbs.yml:1]: Same here need to all config install back with changes for d9

File: composer.lock [hunk 8]
```diff
@@ -0,0 +1,10199 @@
+{
+    "_readme": [
```
Comment (piggito) [inline composer.lock:2]: Please don't include composer.lock file, talked to @davidferlay about it and for now we wan't to let composer always install the latest version of packages according to composer.json constraints

File: composer.json (hunk 1 above)
Comment (andypost) [reply composer.json:2]: Btw the right name is https://packagist.org/packages/skilld-labs/sdc

File: Makefile (hunk 2 above)
Comment (andypost) [reply Makefile:92]: Let's keep changes in own scope, we anyway need working d9-compatible release with DC approach

**Comment by alexkras** [comment]
all changes now in d9 branch

File: composer.json (hunk 1 above)
Comment (andypost) [reply composer.json:2]: Not clear why indent changed from 2 to 4?!

File: composer.json [hunk 9]
```diff
@@ -1,159 +1,116 @@
...
+    "description": "Project template for Drupal 8 projects with composer",
```
Comment (andypost) [inline composer.json:3]: 9!)

File: composer.json [hunk 10]
```diff
@@ -1,159 +1,116 @@
...
-      "web/libraries/{$name}": ["type:drupal-library"],
-      "web/modules/contrib/{$name}": ["type:drupal-module"],
-      "web/profiles/contrib/{$name}": ["type:drupal-profile"],
-      "web/themes/contrib/{$name}": ["type:drupal-theme"],
-      "drush/contrib/{$name}": ["type:drupal-drush"]
+    "require": {
+        "ckeditor/panelbutton": "^4.10",
+        "cweagans/composer-patches": "^1.6.5",
+        "composer/installers": "^1.2",
+        "drupal/core-composer-scaffold": "^9",
```
Comment (andypost) [inline composer.json:34]: it should not be used

Comment (andypost) [reply composer.json:34]: As you can see at https://packagist.org/packages/drupal/core-recommended#9.0.0-beta2
dependencies already require `composer/installers: v1.8.0` and message is unused

File: composer.json [hunk 11]
```diff
@@ -1,159 +1,116 @@
...
-        "Added check to see if TTY is available": "https://patch-diff.githubusercontent.com/raw/mglaman/drupal-check/pull/135.patch"
-      }
+    "minimum-stability": "dev",
+    "prefer-stable": true,
+    "config": {
+        "sort-packages": true
+    },
+    "autoload": {
+        "classmap": [
+            "scripts/composer/ScriptHandler.php"
```
Comment (andypost) [inline composer.json:57]: this tool needs updates too, including patch from https://www.drupal.org/project/drupal/issues/3108262 (will help make releases smaller

File: composer.json [hunk 12]
```diff
@@ -1,159 +1,116 @@
...
-        "Add missing schema for menu_link_attributes": "https://patch-diff.githubusercontent.com/raw/yannickoo/menu_link_attributes/pull/52.patch"
-      },
-      "drupal/svg_image": {
-        "Add missing schema for svg_image": "https://www.drupal.org/files/issues/2018-12-31/missing_schema_field_formatter.patch"
-      },
-      "drupal/focal_point": {
-        "Integrate focal point with media_library": "https://www.drupal.org/files/issues/2019-12-11/focal_point-compatibility-with-media-libary-3094478-2.patch"
-      },
-      "mglaman/drupal-check": {
-        "Added check to see if TTY is available": "https://patch-diff.githubusercontent.com/raw/mglaman/drupal-check/pull/135.patch"
```
Comment (andypost) [inline composer.json:155]: I bet removal if this patch require to upgrade `"mglaman/drupal-check": "^1.1"` statement

File: composer.json [hunk 13]
```diff
@@ -1,159 +1,116 @@
...
-        "Import multiple custom translation po files": "https://patch-diff.githubusercontent.com/raw/drush-ops/drush/pull/4251.patch"
-      },
-      "drupal/lb_ux": {
-        "Bloc edition when Show content preview is disabled": "https://www.drupal.org/files/issues/2020-02-27/3116402.patch"
-      },
-      "drupal/config_inspector": {
-        "DrupalCoreLink could not be converted to string": "https://www.drupal.org/files/issues/2019-10-02/3084666-4.patch"
-      },
-      "drupal/default_content": {
-        "Do not reimport existing entities": "https://www.drupal.org/files/issues/do_not_reimport-2698425-56.patch"
```
Comment (andypost) [inline composer.json:140]: I better keep it for transition time

File: web/profiles/sdd_core/sdd_core.info.yml [hunk 14]
```diff
@@ -1,7 +1,8 @@
 name: sdd_core
 type: profile
 description: 'Skill drupal development installation profile for light projects'
-core: 8.x
+core_version_requirement: ^9
+
```
Comment (andypost) [inline web/profiles/sdd_core/sdd_core.info.yml:5]: nit) please remove xtra empty line

File: web/profiles/sdd/sdd.info.yml (hunk 5 above)
Comment (andypost) [reply web/profiles/sdd/sdd.info.yml:11]: We need to fix any of this modules, or add patches but the list should not be affected

**Comment by piggito** [comment]
Closed in favor of https://github.com/skilld-labs/skilld-docker-container/pull/243

**Pull request closed without merging**